### PR TITLE
KAZOO-4599: Disconnect channels when account credit exhausted

### PR DESCRIPTION
**Pull request by skorobkov**
j5_balance_crawler was moved to ecallmgr.
Periodiacly it get list acccount with `per_minute` calls and ask jonny5 if this accounts balance ok?
If not ok - get list `per_minute` channels and send `hangup` command for this channels.

Added `authz.balance_check` request. Request contain `Accounts` list, which balance should be checked.
```JSON
{
  "Accounts": [
    "30848a1d2baba2165d6373fa961c43a9",
    "37c8ecfc3cbddc3a5d7e83634ae8241d",
    "64c175419824af023d79842d87d9a72e"
  ]
}
```
Response containt list of tuples.
```JSON
{
  "Balances": [
    {
      "30848a1d2baba2165d6373fa961c43a9": true
    },
    {
      "37c8ecfc3cbddc3a5d7e83634ae8241d": false
    },
    {
      "64c175419824af023d79842d87d9a72e": true
    }
  ]
}
```
`true` - balance is ok
`false` - balance is negative, and ecallmgr should to disconnect calls of this account.

### CODE REVIEW

File: applications/ecallmgr/src/ecallmgr_fs_channels.erl [hunk 1]
```diff
@@ -143,6 +145,62 @@ show_all() ->
...
+    MatchSpec = [
+                 {#channel{node = '$1', uuid = '$2', reseller_id = AccountId, reseller_billing = <<"per_minute">>, _ = '_'}
+                   ,[]
+                   ,[{{'$1', '$2'}}]
+                  }
+                 ,{#channel{node = '$1', uuid = '$2', account_id = AccountId, account_billing = <<"per_minute">>, _ = '_'}
+                   ,[]
+                   ,[{{'$1', '$2'}}]
+                  }
+%%                ,{#channel{node = '$1', uuid = '$2', reseller_id = AccountId, _ = '_'}
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_channels.erl]: Remove dead code

File: applications/ecallmgr/src/ecallmgr_fs_channels.erl [hunk 2]
```diff
@@ -143,6 +145,62 @@ show_all() ->
...
+                 }
+                 ,{#channel{reseller_id = '$1', reseller_billing = <<"per_minute">>, _ = '_'}
+                   ,[{'=/=', '$1', 'undefined'}]
+                   ,['$$']
+                  }
+                 ,{#channel{account_id = '$1', account_billing = <<"per_minute">>, _ = '_'}
+                   ,[{'=/=', '$1', 'undefined'}]
+                   ,['$$']
+                  }
+%%                ,{#channel{account_id = '$1', reseller_id = '$2', _ = '_'}
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_channels.erl]: Remove dead code

File: applications/ecallmgr/src/ecallmgr_fs_channels.erl [hunk 3]
```diff
@@ -143,6 +145,62 @@ show_all() ->
...
+%%                  ,['$$']
+%%                 }
+%%                ,{#channel{reseller_id = '$1', _ = '_'}
+%%                  ,[{'=/=', '$1', 'undefined'}]
+%%                  ,['$$']
+%%                 }
+                ],
+    lists:usort(lists:flatten(ets:select(?CHANNELS_TBL, MatchSpec))).
+
+-spec per_minute_channels(api_binary()) -> api_binaries().
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_channels.erl]: Fix spec. I believe return type is incorrect, no? Should be a tuple?

**Comment by skorobkov** [comment]
Is there anything I should do?

**Comment by jamesaimonetti** [comment]
@skorobkov sorry for the delay. I talked with @k-anderson about this at the end of last week and he had some thoughts. This was followed by an unexpected conference call with a client issue which took the rest of our day up. I'll remind him today to see if he can get you some feedback here.

**Pull request merged**